File: backend/src/forms/lrembecki.core.forms/Forms/FormVM.cs
namespace lrembecki.core.forms.Forms;

public record FormVM(
    Guid Id,
    Guid FormDefinitionId,
    DateTime CreatedAt,
    Dictionary<string, string> Fields
)
{
    internal static FormVM Map(FormEntity entity)
    {
        if (entity is null) return null!;

        return new FormVM(
            entity.Id, 
            entity.FormDefinitionId,
            entity.CreatedAt,
            entity.Fields.ToDictionary(e => e.Name, e => e.Value)
        );
    }
}

File: frontend/src/app/modules/forms/form.vm.ts
export class FormVM {
  id: string = null!;
  formDefinitionId: string = null!;
  name: string = null!;
  createdAt: Date = null!;

  constructor(init?: Partial<FormVM>) {
    Object.assign(this, init);

    if (this.createdAt) this.createdAt = new Date(this.createdAt);
  }
}
